Create_Volume_From_Snapshot.md

```diff
@@ -0,0 +1,84 @@
+Create Volumes from Snapshots
+
+**Status**: Pending
+
+**Version**: Alpha
+
+**Author(s)**: Xiaoyan Li (@lixiaoy1)
+
+## Summary
+
+This proposal is drafted for creating a volume from snapshot supported backend storages. It defines the API interfaces to OpenSDS customers and meanwhile the interfaces for kinds of storage drivers.
+
+## Motivation
+
+Under some scenarios users may want to reuse the data on snapshots, with this function they can easily do it.
+
+### Goals
+
+* Design a unified API interface to create volumes from snapshots supported by backend storages.
+* Complete the common logic to create volumes from snapshots, and implementations in LVM, Ceph and Cinder.
+
+### Non-Goals
+
+* The implementations inside storage drivers except LVM, Ceph and Cinder.
+* It does not provide the function to create volumes from snapshots which is not supported by backend storages. For example, creating volumes from snapshots in different storages is not supported.
+
+## Proposal
+
+The proposal is drafted to allow users to create volumes from snapshots.
```

Comment:
This all looks good, but I think it might be nice to expand on the proposal details a little more.

Will different sizes be allowed when creating a volume from a snapshot?
Are there any constraints on how many volumes can be created from a given snapshot?
Is there anything else that changes with the create flow other than just including the snapshot ID?

I'm sure a lot of this would come out in the implementation, but it might be nice to include some more details here so if someone reads this doc later on they get a better understanding of what to expect.

Reply:
I agree with Sean's input here.  Do you need to specify the target volume?  If the target volume is larger do you extend?  I think that would be helpful to set those goals here.

Reply:
Thanks to @stmcginnis and @jungleboyj  good point!  I will make this more detailed.